README.rst

```diff
@@ -393,6 +393,19 @@ If you want to run Nitpick as a flake8 plugin instead::
           - id: flake8
             additional_dependencies: [nitpick]
 
+Run as a MegaLinter plugin
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you use `MegaLinter <https://megalinter.github.io/>`_ you can run Nitpick as a plugin. Add the following two entries to your ``.mega-linter.yml`` configuration file:
+
+.. code-block:: yaml
+
+    PLUGINS:
+      - https://raw.githubusercontent.com/andreoliwa/nitpick/v0.32.0/mega-linter-plugin-nitpick/nitpick.megalinter-descriptor.yml
+    ENABLE_LINTERS:
+      - nitpick
```

Comment:
I wanted to try Megalinter, I might be doing something wrong.

I created `.mega-linter.yml` with on Nitpick's own dir:

```yaml
PLUGINS:
  - https://raw.githubusercontent.com/andreoliwa/nitpick/008b2492da43ea142a182f463117528f9fc45f1b/mega-linter-plugin-nitpick/nitpick.megalinter-descriptor.yml
ENABLE_LINTERS:
  - nitpick
```

I commented out the whole `[isort]` section on `setup.cfg` because I know `nitpick fix` should restore parts of it.

I get a warning on PyCharm and this output (with empty "MATCHING LINTERS" and "SUMMARY" tables):

<details>

![image](https://user-images.githubusercontent.com/1258218/160494576-b7f0d52f-081f-47c5-9297-23e8c3cc98aa.png)

```
(...)
Listing all files in directory [/tmp/lint], then filter with:
- Excluding .gitignored files [24036]: /tmp/lint/.DS_Store, /tmp/lint/.cache/nitpick/0a/c8/93/38/d6/08/cc/7e/0ac89338d608cc7e7d19f21f25086e09bc6cb5307ed0d3503b80d0bb06853780, /tmp/lint/.cache/nitpick/0d/00/3f/9f/5f/bc/7e/f3/0d003f9f5fbc7ef335cf999388040b892a6de5f6b82b5b46400daba99f0c1d3c, /tmp/lint/.cache/nitpick/16/83/8e/a4/61/eb/ed/e4/16838ea461ebede4183ff1140bc509501d745850897e332a6bf672c44f2bc78a, /tmp/lint/.cache/nitpick/17/17/f9/8e/92/63/61/af/1717f98e926361af973cf7bf8eb03376a02b856efc417993674e3d942048445a, /tmp/lint/.cache/nitpick/18/44/7a/f5/85/6e/06/33/18447af5856e063358fe3644951951a09d2ed5d41b90fffdd63f36ecae56f33c, /tmp/lint/.cache/nitpick/25/07/8e/83/7a/b6/ab/f7/25078e837ab6abf713dd9fe5afd4f7fd2627770e1e5a536e032356a479c30328, /tmp/lint/.cache/nitpick/37/75/ee/6f/d2/c6/29/9b/3775ee6fd2c6299bbba801203524c7b0bc19d5d81be1f615deac0ec2dedab1a7, /tmp/lint/.cache/nitpick/38/65/95/34/0c/b1/b6/81/386595340cb1b6817f2bc0043d196bebdc5a4db14f10c82f8301164a19086344, /tmp/lint/.cache/nitpick/3a/68/ac/df/41/f3/eb/71/3a68acdf41f3eb712ba858d3e7df91e403e59de12551f069422d06ee2a5d79ae,...(full list in DEBUG)
Kept [0] files on [15056] found files

+----MATCHING LINTERS-+----------+----------------+------------+
| Descriptor | Linter | Criteria | Matching files | Format/Fix |
+------------+--------+----------+----------------+------------+


+----SUMMARY-+--------+------+-------+-------+--------+--------------+
| Descriptor | Linter | Mode | Files | Fixed | Errors | Elapsed time |
+------------+--------+------+-------+-------+--------+--------------+

You could have same capabilities but better runtime performances if you use a MegaLinter flavor:
- [ci_light] megalinter/megalinter/flavors/ci_light@v5 (16 linters) https://megalinter.github.io/flavors/ci_light/
- [documentation] megalinter/megalinter/flavors/documentation@v5 (41 linters) https://megalinter.github.io/flavors/documentation/
- [dart] megalinter/megalinter/flavors/dart@v5 (42 linters) https://megalinter.github.io/flavors/dart/
- [ruby] megalinter/megalinter/flavors/ruby@v5 (42 linters) https://megalinter.github.io/flavors/ruby/
- [rust] megalinter/megalinter/flavors/rust@v5 (42 linters) https://megalinter.github.io/flavors/rust/
- [scala] megalinter/megalinter/flavors/scala@v5 (42 linters) https://megalinter.github.io/flavors/scala/
- [swift] megalinter/megalinter/flavors/swift@v5 (42 linters) https://megalinter.github.io/flavors/swift/
- [go] megalinter/megalinter/flavors/go@v5 (43 linters) https://megalinter.github.io/flavors/go/
- [java] megalinter/megalinter/flavors/java@v5 (43 linters) https://megalinter.github.io/flavors/java/
- [salesforce] megalinter/megalinter/flavors/salesforce@v5 (44 linters) https://megalinter.github.io/flavors/salesforce/
- [php] megalinter/megalinter/flavors/php@v5 (46 linters) https://megalinter.github.io/flavors/php/
- [terraform] megalinter/megalinter/flavors/terraform@v5 (47 linters) https://megalinter.github.io/flavors/terraform/
- [dotnet] megalinter/megalinter/flavors/dotnet@v5 (48 linters) https://megalinter.github.io/flavors/dotnet/
- [javascript] megalinter/megalinter/flavors/javascript@v5 (50 linters) https://megalinter.github.io/flavors/javascript/
- [python] megalinter/megalinter/flavors/python@v5 (50 linters) https://megalinter.github.io/flavors/python/

[Updated Sources Reporter] No source file has been formatted or fixed
::set-output name=has_updated_sources::0
✅ Successfully linted all files without errors
```

</details>

And `setup.cfg` is not modified. 🤔

Reply:
The schema is too strict; it doesn’t account for extra plugin names. 

You won’t see any fixes as the default mode for MegaLinter is to *only check*; you can enable fixes by [configuring `APPLY_FIXES`]( https://megalinter.github.io/latest/configuration/). That’s because this is designed to run on a CI, not locally.

Reply:
> That’s because this is designed to run on a CI, not locally.

https://pre-commit.ci/ applies fixes to files.

Maybe MegaLinter can be used for that.
Although GitHub Actions would need to commit the files.
I didn't do it so far, don't know if it's possible or easy.

Food for thought. 😉

Reply:
The [MegaLinter installation instructions for GitHub](https://megalinter.github.io/latest/installation/#github-action) includes a workflow that can either create a PR with fixes or just commit directly.

Because we also run pre-commit hooks locally I prefer to just fail the linting step; that way devs learn to ensure their pre-commit setup is in place and not submit PRs that need fixing in the first place ;-D

Reply:
> that way devs learn to ensure their pre-commit setup is in place and not submit PRs that need fixing in the first place 😉

I like this line of thinking. 😅 

I use https://pre-commit.ci/ on my public repos.
But `pre-commit` also runs locally before pushing, so there's nothing left for https://pre-commit.ci/ to fix.